packages/synapse-react-client/src/synapse-queries/entity/useEntityBundle.ts

```diff
@@ -31,6 +33,47 @@ export function useGetEntityBundle<
         version,
         accessToken,
       ),
+  }
+}
+
+export function useGetEntityBundle<
+  T extends EntityBundleRequest = typeof ALL_ENTITY_BUNDLE_FIELDS,
+  TSelect = EntityBundle<T>,
+>(
+  entityId: string,
+  version?: number,
+  bundleRequest: T = ALL_ENTITY_BUNDLE_FIELDS as T,
+  options?: Partial<
+    UseQueryOptions<EntityBundle<T>, SynapseClientError, TSelect>
+  >,
+) {
+  const queryOptions = useGetEntityBundleQueryOptions(
+    entityId,
+    version,
+    bundleRequest,
+  )
+  return useQuery<EntityBundle<T>, SynapseClientError, TSelect>({
+    ...options,
+    ...queryOptions,
+  })
+}
+
+export function useSuspenseGetEntityBundle<
```

Comment:
Get an entity bundle but using [useSuspenseQuery](https://tanstack.com/query/latest/docs/framework/react/guides/suspense#suspense)

Reply:
This is a clean pattern, do you recommend we start using this across the library?

Reply:
I think in many cases, it's fantastic, and 100% worth adopting!

We cannot use it in cases where the query is dependent (i.e. the `enabled` option may be `false`), and it can cause pretty bad request waterfalls if we have many nested components and poorly-targeted `Suspense` wrappers (since nested components do not mount until the parents' suspense queries resolve) . In the latter cases, I think the react-query docs recommend pre-fetching data into the cache as a way to avoid long delays.